### ISSUE TITLE
MeshFunction.mark is not up to date with #623

### ISSUE THREAD

**Issue by francesco-ballarin** [FEniCS/dolfinx#638]
After updating to #623 I went through my code and transposed the point storage. However, is seems that points passed to MeshFunction.mark still use the old layout, i.e. requiring to use `x[:, 1]` rather than `x[1]` in the python layer.

Example to reproduce
```
from numpy import array, finfo, isclose
from ufl import Measure
from dolfin import *
from dolfin.fem import assemble_scalar

mesh = RectangleMesh(MPI.comm_world, [array([-1.0, -2.0, 0.0]), array([1.0, 2.0, 0.0])], [50, 100])

subdomains = MeshFunction("size_t", mesh, 2, 0)
def subdomain(x):
    return x[1] < finfo(float).eps
subdomains.mark(subdomain, 1)

dx = Measure("dx", domain=mesh, subdomain_data=subdomains)

area1 = MPI.sum(mesh.mpi_comm(), assemble_scalar(1.*dx(1)))
print("area1 = ", area1)
assert isclose(area1, 4.)
```

**Comment by garth-wells** [comment]
We must be missing some test coverage for this.

**Comment by michalhabera** [comment]
@garth-wells There are some, they just used the old `x[:, ...]` syntax which stayed consistent with cpp code.

**Comment by michalhabera** [comment]
@francesco-ballarin Should be now fixed with #639 , thanks for spotting this!
